scripts/run-ios-device.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+set -euo pipefail
+set -m # needed to access jobs
+
+GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
+
+# We run Metro in background while calling adb.
+cleanupMetro() {
+    pkill -f run-metro.sh
+    rm -f metro-server-logs.log
+}
+
+# Using function gives a neater jobspec name.
+runMetro() {
+   nohup "${GIT_ROOT}/scripts/run-metro.sh" 2>&1 \
+        | tee metro-server-logs.log
+}
+
+waitForMetro() {
+    set +e # Allow grep command to fail in the loop.
+    TIMEOUT=5
+    echo "Waiting for Metro server..." >&2
+    while ! grep -q "Welcome to Metro" metro-server-logs.log; do
+      echo -n "." >&2
+      sleep 1
+      if ((TIMEOUT == 0)); then
+        echo -e "\nMetro server timed out, exiting" >&2
+        set -e # Restore errexit for rest of script.
+        return 1
+      fi
+      ((TIMEOUT--))
+    done
+    set -e # Restore errexit for rest of script.
+}
+
+# find the first connected iPhone's UUID
+DEVICE_UUID=$(idevice_id -l)
+
+# Check if any device is connected
+if [ -z "$DEVICE_UUID" ]; then
+    echo "No connected iPhone device detected."
+    exit 1
+else
+    echo "Connected iPhone UDID: $DEVICE_UUID"
+fi
+
+BUILD_DIR="${GIT_ROOT}/build"
+
+#iOS build of debug scheme
+xcodebuild -workspace "ios/StatusIm.xcworkspace" -configuration Debug -scheme StatusIm -destination id="$DEVICE_UUID" -derivedDataPath "${BUILD_DIR}" | xcbeautify
+
+APP_PATH="${BUILD_DIR}/Build/Products/Debug-iphoneos/StatusIm.app"
+
+# Install on the connected device
+DEVICECTL_OUTPUT=$(xcrun devicectl device install app --device "$DEVICE_UUID" "$APP_PATH" 2>&1 | tee >(cat 1>&2))
+
+# Extract installationURL
+INSTALLATION_URL=$(echo "$DEVICECTL_OUTPUT" | grep "installationURL:" | sed -E 's/.*installationURL: (.*)/\1/')
+
+trap cleanupMetro EXIT ERR INT QUIT
+runMetro &
+waitForMetro
+
+# wait a bit after installation for the iPhone to calm down
+sleep 3
+
```

Comment:
sometimes the installed app would insta crash, 
This sleep was placed to test that.
The app sometimes still crashes, you can re-open it by tapping on the icon, not ideal DX however

Reply:
fixed with : https://github.com/status-im/status-mobile/pull/18845/commits/4e538e39e686bc72c5c5643b6dc8c2106d24d459